File: packages/jsii-rosetta/test/translations/imports/submodule-import.cs
using Amazon.JSII.Tests.CalculatorNamespace;
using Amazon.JSII.Tests.CalculatorNamespace.HomonymousForwardReferences;
using Gen.Providers.Aws;

// Access without existing type information
var awsKmsKeyExamplekms = new Kms.KmsKey(this, "examplekms", new Struct {
    DeletionWindowInDays = 7,
    Description = "KMS key 1"
});

// Accesses two distinct points of the submodule hierarchy
var myClass = new Submodule.MyClass(new SomeStruct { Prop = Submodule.Child.SomeEnum.SOME });

// Access via a renamed import
Foo.Consumer.Consume(new ConsumerProps { Homonymous = new Homonymous { StringProperty = "yes" } });

File: packages/jsii-rosetta/test/translations/imports/submodule-import.java
import software.amazon.jsii.tests.calculator.submodule.*;
import software.amazon.jsii.tests.calculator.submodule.child.*;
import software.amazon.jsii.tests.calculator.homonymousForwardReferences.*;
import software.amazon.jsii.tests.calculator.homonymousForwardReferences.foo.*;
import gen.providers.aws.kms.*;

// Access without existing type information
Object awsKmsKeyExamplekms = KmsKey.Builder.create(this, "examplekms")
        .deletionWindowInDays(7)
        .description("KMS key 1")
        .build();

// Accesses two distinct points of the submodule hierarchy
MyClass myClass = MyClass.Builder.create().prop(SomeEnum.SOME).build();

// Access via a renamed import
Consumer.consume(ConsumerProps.builder().homonymous(Homonymous.builder().stringProperty("yes").build()).build());
